irc/client_lookup_set.go

```diff
@@ -207,6 +207,22 @@ func (clients *ClientManager) AllWithCaps(capabs ...caps.Capability) (sessions [
 	return
 }
 
+// AllWithCapsNotify returns all clients with the given capabilities, and that support cap-notify.
+func (clients *ClientManager) AllWithCapsNotify(capabs ...caps.Capability) (sessions []*Session) {
+	clients.RLock()
+	defer clients.RUnlock()
+	for _, client := range clients.byNick {
+		for _, session := range client.Sessions() {
+			capabs = append(capabs, caps.CapNotify)
```

Comment:
This appends an extra copy of `caps.CapNotify` on every iteration, right?

Reply:
... good point